File: week-2_arrays/3-Problem_set/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// Take imput from user (text:) in main

// Function to count letters in text
int count_letters(string text);

// Function to count words
int count_words(string text);

// Function to count sentences
int count_sentences(string text);

// Function for Coleman Liau index
int index(int words, int letters, int sentences);

// Function for Coleman Liau index = 0.0588 * L - 0.296 * S - 15.8
//      where L = letters per 100 words   and   S = sentences per 100 words

//Print output in main: "Grade X" where x=1-16, "Grade 16+" or "Before Grade 1"
int main(void)
{
    string text = get_string("Text: ");
    int w = count_words(text);
    float l = count_letters(text) * 100.0 / w;
    float s = count_sentences(text) * 100.0 / w;
    int index = round((0.0588 * l) - (0.296 * s) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}

int count_letters(string text)                           // Function to count letters in text
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if isalpha(text[i])
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)                            // Function to count words
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if isspace(text[i])
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)                        // Function to count sentences
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: week-6_python/3-Problem_set/sentimental-readability/readability.py
# TODO

import cs50


# Main function
def main():
    text = cs50.get_string("Text: ")
    w = count_words(text)
    l = count_letters(text) * 100.0 / w
    s = count_sentences(text) * 100.0 / w
    index = round((0.0588 * l) - (0.296 * s) - 15.8)

    if (index < 1):
        print("Before Grade 1")

    elif (index > 16):
        print("Grade 16+")

    else:
        print(f"Grade {index}")


# Function to count letters in text
def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha() == True:
            letters += 1

    return letters


# Function to count words
def count_words(text):
    words = 1
    for i in range(len(text)):
        if text[i] == " " or text[i] == "   ":
            words += 1

    return words


# Function to count sentences
def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1

    return sentences


# Call main
main()